src/useNavigationBuilder.tsx

```diff
@@ -49,10 +51,13 @@ const getRouteConfigsFromChildren = <ScreenOptions extends object>(
     );
   }, []);
 
-export default function useNavigationBuilder<ScreenOptions extends object>(
-  router: Router<any>,
-  options: Options
-) {
+export default function useNavigationBuilder<
+  ScreenOptions extends object,
+  Options extends DefualtOptions = DefualtOptions
+>(router: RouterHelper<any, Options>, options: Options) {
+  const { current: currentRouter } = React.useRef<Router<any>>(
+    typeof router === 'function' ? router(options) : router
```

Comment:
I think it's better to just always make it a function rather than accepting both object and function

Reply:
I think we shouldn't pass things like `children` to the router. It'll be problematic if someone writing a custom router uses children.